scripts/readme.js

```diff
@@ -1,70 +1,98 @@
 import fs from 'fs';
+import argv from 'minimist';
 import path from 'path';
 import prettier from 'prettier';
 
-const readme = fs.readFileSync('docs/index.md', 'utf-8');
+const SOURCE = 'docs/index.md';
 
-const headers = readme.match(/(?<!\S)(#+)\s*([^\n]+)/g);
-const table_of_contents = headers
-	.map((h) =>
-		h.replace(/^(#+)\s*(.*)/, (m, prefix, label) => {
-			const indent = '  '.repeat(prefix.length - 2);
-			return `${indent}- [${label}](#${label
-				.toLowerCase()
-				.replace(/\s/g, '-')})`;
-		})
-	)
-	.join('\n');
+// eslint-disable-next-line no-undef
+const args = argv(process.argv.slice(2), {
+	boolean: true,
+	alias: { w: 'watch' },
+});
 
-const result = readme
-	.replace(/\[\[toc\]\]/, () => table_of_contents)
-	.replace(
-		/<<< @((\/[^/ #\s]+)+)(#(\S+))?( \[(\S+)\])?/g,
-		(_m, filepath, _base, _r, region, _l, tab) => {
-			const ext = path.extname(filepath);
-			const resolved = path.join('./docs', `.${filepath}`);
-			let result = fs.readFileSync(resolved);
+if (args.watch) {
+	// eslint-disable-next-line no-undef
+	console.log(`Watching ${SOURCE} for changes...`);
+	fs.watch(SOURCE, async () => {
+		const updated = await updateIfDifferent();
+		if (updated) {
+			// eslint-disable-next-line no-undef
+			console.log('README.md updated!');
+		}
+	});
+} else {
+	const updated = await updateIfDifferent();
+	// eslint-disable-next-line no-undef
+	if (updated) process.exit(1);
+}
+
+async function updateIfDifferent() {
+	const readme = fs.readFileSync(SOURCE, 'utf-8');
+
+	const headers = readme.match(/(?<!\S)(#+)\s*([^\n]+)/g);
+	const table_of_contents = headers
+		.map((h) =>
+			h.replace(/^(#+)\s*(.*)/, (m, prefix, label) => {
+				const indent = '  '.repeat(prefix.length - 2);
+				return `${indent}- [${label}](#${label
+					.toLowerCase()
+					.replace(/\s/g, '-')})`;
+			})
+		)
+		.join('\n');
+
+	const result = readme
+		.replace(/\[\[toc\]\]/, () => table_of_contents)
+		.replace(
+			/<<< @((\/[^/ #\s]+)+)(#(\S+))?( \[(\S+)\])?/g,
+			(_m, filepath, _base, _r, region, _l, tab) => {
+				const ext = path.extname(filepath);
+				const resolved = path.join('./docs', `.${filepath}`);
```

Comment:
When I work on my windows machine, this generates another path.
Instead of using `/` as the path separator it uses `\`, which results in changes that will break during a PR.
This is something we'll need to take a look at because it might hinder people from pushing changes.

Reply:
I think it's just a matter of removing the `/` from `./docs`. The `path` utility should already handle system specific separators. I updated the code. Give it a shot and let me know if it works for you.

Reply:
I saw your code and made another change on top of it.

```ts
const WRONG_SEPARATOR =
	path.sep === path.win32.sep ? path.posix.sep : path.win32.sep;
const WRONG_SEPARATOR_RGX = new RegExp('\\' + WRONG_SEPARATOR, 'g');

export function stringToPath(str) {
	return str.replace(WRONG_SEPARATOR_RGX, path.sep);
}
```

This will actually take into account which system you're on and adjust the path accordingly.